scripts/xenadmin-build.ps1

```diff
@@ -0,0 +1,282 @@
+# Copyright (c) Cloud Software Group, Inc.
+#
+# Redistribution and use in source and binary forms,
+# with or without modification, are permitted provided
+# that the following conditions are met:
+#
+# *   Redistributions of source code must retain the above
+#     copyright notice, this list of conditions and the
+#     following disclaimer.
+# *   Redistributions in binary form must reproduce the above
+#     copyright notice, this list of conditions and the
+#     following disclaimer in the documentation and/or other
+#     materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND
+# CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+# BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+# NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+# SUCH DAMAGE.
+
+Param(
+    [Parameter(HelpMessage = " Global build number")]
+    [int]$buildNumber
+)
+
+$ErrorActionPreference = "Stop"
+
+function mkdir_clean([string]$path) {
+    if ([System.IO.Directory]::Exists($path)) {
+        Remove-Item -Path $path -Force -Recurse
+    }
+    New-Item -ItemType Directory -Path $path -Verbose
+}
+
+function build([string]$solution) {
+    msbuild /m /verbosity:minimal /p:Configuration=Release /p:TargetFrameworkVersion=v4.8 /p:VisualStudioVersion=17.0 $solution
+}
+
+function get_locale_id([string]$locale) {
+    switch ($locale) {
+        "ja-jp" { 1041 }
+        "zh-cn" { 2052 }
+        "zh-tw" { 1028 }
+        default { 1033 } #en-us
+    }
+}
+
```

Comment:
again very nit picky (sorry!) so feel free to ignore.

Can we have some nice `#region`s? visually it helps navigate the script.

We could also have sections using more visible titles such as the ones below.

```powershell
############################################################
#                    Helper functions                      #
############################################################

function get_locale_id([string]$locale) {
    switch ($locale) {
        ...
    }
}

############################################################
#                    Sign deliverables                     #
############################################################

if ([System.IO.File]::Exists("$REPO\scripts\sign.ps1")) {
   ...
}
else {
    Write-Host "Sign script does not exist; skip signing binaries"
}

```

Reply:
I thnk I'll go for title visibility as regions are not understood by all environments (for example, they work in ISE, but VS Code doesn't seem to understand them).